pages/deep_well_plates.vue

```diff
@@ -0,0 +1,168 @@
+}
+interface ResultStatus {
+  status: string
+  ready: string
+  _rowVariant: string
+}
+interface ResultRow extends ResultStatus {
+  barcode: string
+}
+
+export default defineComponent({
+  name: 'DeepWellPlates',
+  data() {
+    return {
+      status: statuses.Idle,
+      plateBarcodes: '',
+      resultFields: [
+        { key: 'barcode', label: 'Plate Barcode' },
+        { key: 'status', label: 'Status / Errors' },
+        { key: 'ready', label: 'Ready for Limber' },
+      ],
+      results: Array<ResultRow>(),
+    }
+  },
+  methods: {
+    isBusy(): boolean {
+      return this.status === statuses.Busy
+    },
+    isValid(): boolean {
+      return this.parsedBarcodes().length > 0
+    },
+    clearBarcodes() {
+      this.plateBarcodes = ''
+    },
+    async submitBarcodes() {
+      this.status = statuses.Busy
+      const barcodes = this.parsedBarcodes()
+      const responses = await lighthouseService.createPlatesFromBarcodes({
+        barcodes,
+        type: 'rvi_deep_well_96',
+      })
+      this.handleSubmissionResponses(barcodes, responses)
+      this.status = statuses.Idle
+    },
+    /**
+     * Parse barcodes by splitting on white space, throwing away empty values and
+     * then removing duplicates from the list.
+     */
+    parsedBarcodes() {
+      const listNoBlanks = this.plateBarcodes.split(/\s+/).filter((b) => b !== '')
+      return [...new Set(listNoBlanks)]
+    },
+    createStatus(response: Response): ResultStatus {
+      try {
+        if (response.data.value !== null) {
+          return {
+            status: 'Plate was imported successfully.',
+            ready: 'Yes',
+            _rowVariant: 'success',
+          }
+        }
+
+        if (response.error?.value !== null) {
+          if (response.error.value.statusCode === 500) {
+            return {
+              status: 'There was a problem processing the request; please try again.',
+              ready: 'Unknown',
+              _rowVariant: 'warning',
+            }
+          }
+
+          if (response.error.value.data?.errors !== null) {
+            const errors = response.error.value.data.errors.join('; ')
+            if (/is already in use\.$/.test(errors)) {
+              return {
+                status: 'The barcode already exists in Sequencescape.',
+                ready: 'Yes',
+                _rowVariant: 'success',
+              }
+            } else {
+              return { status: `Errors:  ${errors}`, ready: 'No', _rowVariant: 'danger' }
+            }
+          }
+        }
+      } catch {
+        // Fall into the unhandled response below
+      }
+
+      return {
+        status: 'Unhandled response received; please try again.',
+        ready: 'Unknown',
+        _rowVariant: 'warning',
+      }
+    },
+    handleSubmissionResponses(barcodes: string[], responses: Response[]) {
+      this.results = barcodes.map((barcode, idx) => {
```

Comment:
Feels a bit weird rejoining barcodes and responses back together here based on index. Not sure what the alternative here would be though unless the barcodes are included in the response or something.

Reply:
It's a tricky one.  I agree it isn't ideal but the responses do not have any sort of consistent key about which request they came from.  Especially if they're an error condition such as a 500 status code.  The other alternative I was thinking of, but didn't do because it felt like the value wasn't there, was to wrap the requests in objects that also have the barcode that was used to initiate them, then iterate over those.  I still could do that if needed, but I confirmed that the method shown here is working.

Reply:
Ok yep happy to keep this method.